packages/permissions/src/utils/has-permissions.ts

```diff
@@ -242,20 +244,27 @@ export const hasAppliedDataFence = (options: TOptionsForAppliedDataFence) =>
       demandedDataFence.group
     );
 
-    if (actualDataFenceByPermissionGroup) {
-      return Object.entries(actualDataFenceByPermissionGroup).every(
-        ([dataFenceName, dataFenceValue]) => {
-          if (dataFenceValue) {
-            return hasDemandedDataFence({
-              actualDataFence: { name: dataFenceName, dataFenceValue },
-              demandedDataFence,
-              selectDataFenceData: options.selectDataFenceData,
-            });
-          }
-          return false;
-        }
-      );
-    }
+    if (!actualDataFenceByPermissionGroup) return false;
 
-    return false;
+    // we try to find if the demanded dataFence is available inside the actual datafences
+    const specificActualDataFence = Object.entries(
+      actualDataFenceByPermissionGroup
+    ).find(
+      ([dataFenceName, dataFenceValue]) =>
+        dataFenceValue && dataFenceName === toCanCase(demandedDataFence.name)
+    );
+
+    if (!specificActualDataFence) return false;
+
+    const [dataFenceName, dataFenceValue] = specificActualDataFence;
+    return hasDemandedDataFence({
+      actualDataFence: {
+        name: dataFenceName,
+        // we do the type casting because at this point we are sure that
+        // specificActualDataFence.dataFenceValue is not null
+        dataFenceValue: dataFenceValue as { values: string[] },
```

Comment:
Yeah I think it's fine like this for now. You can simplify this by doing it like:

```ts
dataFenceValue as TActualDataFence['dataFenceValue']
```